syntax = "proto3";

message TaskRequest {
  string code_path = 1; // Pantheon path where the input code is stored.
  string data_path = 2; // Pantheon path where the input files are stored.
  string output_path = 3; // Pantheon path where the output will be stored.
  string target_path = 4; // Path to file from code root that is to be executed.
  double timeout = 5; // Timeout for the execution.
  repeated ConfigPair config_pairs = 6; // Key value pairs containing action and related parameters.
  int32 number_of_retries = 7; // Number of times teh request will be sent again in case of failure.
}

// Key value pairs containing action and related parameters.
message ConfigPair {
  string key = 1;
  string value = 2;
}

message TaskResponse {
  enum options {
    SUCCESS = 0;
    FAILURE = 1;
    BUSY = 2;
  }
  options status = 1;    
  int32 number_of_files = 2;
  double time_taken = 3;
}
 